pygame/math.py

```diff
@@ -0,0 +1,1614 @@
+# pygame_cffi - a cffi implementation of the pygame library
+# Copyright (C) 2016  Anton Joubert
+# Copyright (C) 2016  Simon Cross
+# Copyright (C) 2016  Neil Muller
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Library General Public
+# License along with this library; if not, write to the Free
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+# MA  02110-1301  USA
+
+""" Math module """
+
+from __future__ import absolute_import as _absolute_import
+from __future__ import division as _division
+import math as _math
+from numbers import Number as _Number
+
+from pygame.compat import string_types as _string_types
+from pygame.compat import pow_compat as _pow_compat
+
+_VECTOR_EPSILON = 1e-6  # For equality tests (default for new vectors)
+
+
+def enable_swizzling():
+    """Globally enables swizzling for vectors.
+
+    Enables swizzling for all vectors until disable_swizzling() is called.
+    By default swizzling is disabled.
+
+    Note:  There may be performance degradation, as this replaces __getattr__
+           and __setattr__.  Only enable if required.
+    """
+    # __getattr__ is not defined by default, so nothing to save
+    Vector2.__getattr__ = Vector2.__getattr_swizzle__
+    Vector3.__getattr__ = Vector3.__getattr_swizzle__
+
+    if '__oldsetattr__' not in dir(Vector2):
```

Comment:
To enable swizzling we need to replace both `__getattr__` and `__setattr__`.  When we disable we are deleting the replacement `__getattr__`, and restoring the old `__setattr__`.  Does that help?